tests/serverpod_test_server/test_integration/database_operations/transactions/transaction_test.dart

```diff
@@ -143,4 +143,53 @@ void main() async {
     expect(fetchedData.elementAtOrNull(0)?.number, data.number);
     expect(fetchedData.elementAtOrNull(1)?.number, data3.number);
   });
+
+  test(
+      'Given a nested transaction when inserting and then fetching row then the row is found.',
+      () async {
+    var data = UniqueData(number: 1, email: 'test@serverpod.dev');
+
+    await session.db.transaction<void>(
+      (transaction) async {
+        await session.db.transaction<void>(
+          (nestedTransaction) async {
+            await UniqueData.db.insertRow(
+              session,
+              data,
+              transaction: nestedTransaction,
+            );
+          },
+        );
+      },
+    );
+
+    var fetchedData = await UniqueData.db.find(session);
+    expect(fetchedData, hasLength(1));
+    expect(fetchedData.elementAtOrNull(0)?.number, data.number);
+  });
+
+  test(
```

Comment:
Suggestion: Could we add a test that validate that "both" transactions are not canceled when calling cancel on the inner transaction?

Reply:
Added additional test: [bc2badf](https://github.com/serverpod/serverpod/pull/2656/commits/bc2badf01ab75e040b9c81693b49663b6fc9dcaf)